packages/components/src/swatch/index.tsx

```diff
@@ -7,8 +7,9 @@ import { swatch } from '@wordpress/icons';
  * Internal dependencies
  */
 import Icon from '../icon';
+import type { SwatchProps } from './types';
 
-function Swatch( { fill } ) {
+function Swatch( { fill }: SwatchProps ) {
```

Comment:
Oof, another surprise component I didn't know about 😂 I think we'll need to consolidate this with `ColorIndicator`. Good thing `Swatch` isn't exported in the public package though.

Could we add a warning comment like this?

```suggestion
// This component will be deprecated. Use `ColorIndicator` instead.
// TODO: Consolidate this component with `ColorIndicator`.
function Swatch( { fill }: SwatchProps ) {
```

cc @ciampo

Reply:
Didn't know about this component either 😅 

Some time ago I had drafted a [plan of action](https://github.com/WordPress/gutenberg/issues/33820) to tidy up color-related components, but we never had the resources to actually work on that plan

Reply:
Yep. I think this component can be deprecated. Looks like its only are used in a couple of places (`DuotonePicker` and `ColorListPicker`)

Reply:
Yup — we can definitely deprecate it and even re-implement it by using `ColorIndicator` under the hood.